projects/plugins/jetpack/modules/subscriptions/subscribe-floating-button/class-jetpack-subscribe-floating-button.php

```diff
@@ -0,0 +1,196 @@
+<?php
+/**
+ * Adds support for Jetpack floating Subscribe button feature
+ *
+ * @package automattic/jetpack-subscriptions
+ * @since $$next-version$$
+ */
+
+/**
+ * Jetpack_Subscribe_Floating_Button class.
+ */
+class Jetpack_Subscribe_Floating_Button {
+	/**
+	 * Jetpack_Subscribe_Floating_Button singleton instance.
+	 *
+	 * @var Jetpack_Subscribe_Floating_Button|null
+	 */
+	private static $instance;
+
+	/**
+	 * Jetpack_Subscribe_Floating_Button instance init.
+	 */
+	public static function init() {
+		if ( self::$instance === null ) {
+			self::$instance = new Jetpack_Subscribe_Floating_Button();
+		}
+
+		return self::$instance;
+	}
+
+	const BLOCK_TEMPLATE_PART_SLUG = 'jetpack-subscribe-floating-button';
+
+	/**
+	 * Jetpack_Subscribe_Floating_Button class constructor.
+	 */
+	public function __construct() {
+		if ( get_option( 'jetpack_subscribe_floating_button_enabled', false ) ) {
+			add_action( 'wp_enqueue_scripts', array( $this, 'enqueue_assets' ) );
+			add_action( 'wp_footer', array( $this, 'add_subscribe_floating_button_to_frontend' ) );
+		}
+
+		add_filter( 'get_block_template', array( $this, 'get_block_template_filter' ), 10, 3 );
+
+		add_filter(
+			'jetpack_options_whitelist',
+			function ( $options ) {
+				$options[] = 'jetpack_subscribe_floating_button_enabled';
+
+				return $options;
+			}
+		);
+	}
+
+	/**
+	 * Returns the block template part ID.
+	 *
+	 * @return string
+	 */
+	public static function get_block_template_part_id() {
+		return get_stylesheet() . '//' . self::BLOCK_TEMPLATE_PART_SLUG;
+	}
+
+	/**
+	 * Makes get_block_template return the WP_Block_Template for the floating Subscribe button.
+	 *
+	 * @param WP_Block_Template $block_template The block template to be returned.
+	 * @param string            $id Template unique identifier (example: theme_slug//template_slug).
+	 * @param string            $template_type Template type: `'wp_template'` or '`wp_template_part'`.
+	 *
+	 * @return WP_Block_Template|null
+	 */
+	public function get_block_template_filter( $block_template, $id, $template_type ) {
```

Comment:
As a follow up, we should probably have a parent class that implements these so we don't need to reimplement each time.

Reply:
Yasss: 
- https://github.com/Automattic/jetpack/issues/39853